dev/src/audio.cpp

```diff
@@ -22,31 +22,54 @@ using namespace lobster;
 
 void AddSound(NativeRegistry &nfr) {
 
-nfr("play_wav", "filename,loops", "SI?", "I",
+nfr("play_wav", "filename,loops,prio", "SI?I?", "I",
     "plays a sound defined by a wav file (RAW or MS-ADPCM, any bitrate other than 22050hz 16bit"
     " will automatically be converted on first load). the default volume is the max volume (1.0)"
     " loops is the number of repeats to play (-1 repeats endlessly, omit for no repeats)."
+    " prio is the priority of the sound which determines whether it can be deleted or not"
+    " in case of too many play function calls."
     " returns the assigned channel number (1..8) or 0 on error",
-    [](StackPtr &, VM &, Value &ins, Value &loops) {
-        int ch = SDLPlaySound(ins.sval()->strv(), false, 1.0, loops.intval());
+    [](StackPtr &, VM &, Value &ins, Value &loops, Value &prio) {
+        int ch = SDLPlaySound(ins.sval()->strv(), false, 1.0, loops.intval(), prio.intval());
         return Value(ch);
     });
 
-nfr("play_sfxr", "filename,loops", "SI?", "I",
+nfr("play_sfxr", "filename,loops,prio", "SI?I?", "I",
     "plays a synth sound defined by a .sfs file (use http://www.drpetter.se/project_sfxr.html"
     " to generate these). the default volume is the max volume (1.0)"
     " loops is the number of repeats to play (-1 repeats endlessly, omit for no repeats)."
+    " prio is the priority of the sound which determines whether it can be deleted or not"
+    " in case of too many play function calls."
     " returns the assigned channel number (1..8) or 0 on error",
-    [](StackPtr &, VM &, Value &ins, Value &loops) {
-        int ch = SDLPlaySound(ins.sval()->strv(), true, 1.0, loops.intval());
+    [](StackPtr &, VM &, Value &ins, Value &loops, Value &prio) {
+        int ch = SDLPlaySound(ins.sval()->strv(), true, 1.0, loops.intval(), prio.intval());
         return Value(ch);
     });
 
+nfr("sound_channels", "", "", "I",
```

Comment:
I don't think we want this and the following function.. I'd rather sound management be full automatic once you choose good priorities

Reply:
I do agree re: `sound_channels`. Regarding the `sound_status` function, isn't that useful to be able to get the channel (is it paused/playing etc.)?

Reply:
I'm not sure when that would be useful, but left it in for now